### PR TITLE
feat: delete logs older than X seconds

### PR DESCRIPTION
**Pull request by christyjacob4**
Added function to delete logs older than X seconds

### CODE REVIEW

File: src/Abuse/Adapters/ReCaptcha.php [hunk 1]
```diff
@@ -79,4 +79,28 @@ public function check()
 
         return $result['success'];
     }
+
+    /**
+     * Delete logs older than $seconds seconds
+     *
```
Comment (christyjacob4) [inline src/Abuse/Adapters/ReCaptcha.php:85]: Is it okay to leave these methods unimplemented? What do you suggest @eldadfux

File: src/Abuse/Adapters/ReCaptcha.php [hunk 2]
```diff
@@ -79,4 +79,28 @@ public function check()
...
     }
+
+    /**
+     * Delete logs older than $seconds seconds
+     * 
+     * @param int $seconds 
+     * 
+     * @return bool   
+     */
+    public function deleteLogsOlderThan(int $seconds):bool
```
Comment (eldadfux) [inline src/Abuse/Adapters/ReCaptcha.php:90]: Throw an exception saying this methods are not supported for the Recaptch adapter.

File: src/Abuse/Adapters/TimeLimit.php [hunk 3]
```diff
@@ -202,6 +202,43 @@ protected function hit(string $key, int $time): void
         $this->count++;
     }
 
+    /**
+     * Get all abuse logs
+     *
+     * Returns all the abuse logs that are currently in the DB
+     *
+     * @return array
+     */
```
Comment (eldadfux) [inline src/Abuse/Adapters/TimeLimit.php:211]: Maybe allow to pass a limit and offset here, and just call getLogs. Might not be a very smart idea to use this method on large DB sets.

File: src/Abuse/Adapter.php [hunk 4]
```diff
@@ -12,4 +12,23 @@ interface Adapter
...
     public function check();
+
+    /**
+     * Get all abuse logs
+     *
+     * Returns all the abuse logs that are currently in the DB
+     *
+     * @return array
+     */
+    public function getAllLogs(): array;
```
Comment (eldadfux) [inline src/Abuse/Adapter.php:23]: Maybe allow to pass a limit and offset here, and just call getLogs. Might not be a very smart idea to use this method on large DB sets.

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: src/Abuse/Adapter.php (hunk 4 above)
Comment (christyjacob4) [reply src/Abuse/Adapter.php]: Done

File: src/Abuse/Adapters/ReCaptcha.php (hunk 2 above)
Comment (christyjacob4) [reply src/Abuse/Adapters/ReCaptcha.php:92]: Done

File: src/Abuse/Adapters/TimeLimit.php (hunk 3 above)
Comment (christyjacob4) [reply src/Abuse/Adapters/TimeLimit.php:214]: Done

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
Great work @christyjacob4 🤟

**Pull request merged**